src/main/java/com/feedzai/commons/sql/abstraction/batch/AbstractBatch.java

```diff
@@ -58,11 +57,23 @@
     protected static final int salt = 100;
 
     /**
-     * Lock used for concurrent access to the flush buffer
+     * Lock used for concurrent access to the flush buffer.
      *
      * @since 2.1.4
      */
-    private final Lock flushLock = new ReentrantLock();
+    private final Lock bufferLock = new ReentrantLock();
+
+    /**
+     * Lock used for concurrent access to the flush transaction.
+     * <p>
+     * We are not using the {@link #bufferLock} because the
```

Comment:
SSorry, because of the highlight I was reading the flushTransactionLock javadoc as it would be bufferLock